execution_workflows/QAPA/bin/check_samplesheet.py

```diff
@@ -0,0 +1,136 @@
+#!/usr/bin/env python
+
+import os
+import sys
+import argparse
+
+def parse_args(args=None):
+    Description = "Reformat APAeval samplesheet file and check its contents."
+    Epilog = "Example usage: python check_samplesheet.py <FILE_IN> <FILE_OUT>"
+
+    parser = argparse.ArgumentParser(description=Description, epilog=Epilog)
+    parser.add_argument("FILE_IN", help="Input samplesheet file.")
+    parser.add_argument("FILE_OUT", help="Output file.")
+    return parser.parse_args(args)
+
+
+def make_dir(path):
+    if len(path) > 0:
+        try:
+            os.makedirs(path)
+        except OSError as exception:
+            if exception.errno != errno.EEXIST:
+                raise exception
+
+
+def print_error(error, context='Line', context_str=''):
+    error_str = "ERROR: Please check samplesheet -> {}".format(error)
+    if context != '' and context_str != '':
+        error_str = "ERROR: Please check samplesheet -> {}\n{}: '{}'".format(error, context.strip(), context_str.strip())
+    print(error_str)
+    sys.exit(1)
+
+
+def check_samplesheet(file_in, file_out):
+    """
+    This function checks that the samplesheet follows the following structure:
+    sample,fastq1,fastq2,bam,bai,gff,fasta,bed,mart_export
+    """
+
+    input_extensions = []
+    sample_info_list = []
+    with open(file_in, "r") as fin:
+
+        ## Check header
+        MIN_COLS = 8
+        HEADER = ['sample','fastq1','fastq2','bam', 'bai','gff','fasta','bed','mart_export']
```

Comment:
What does the Biomart file contain? I'm not sure this is part of our set of standardized input files. Is there perhaps a way of generating this file from the standard input set programmatically?